src/components/NavBar/NavBar.component.jsx

```diff
@@ -1,33 +1,40 @@
 import React, { useState } from 'react';
+import { useGlobal } from '../../providers/Global';
 import { API_BASE_URL } from '../../utils/constants';
+import { useFetch } from '../../utils/hooks/useFetch';
 import { NavBar as Navigation, SearchFieldWrapper, SearchField, CheckBox, CheckBoxLabel, CheckBoxWrapper, LoginButton, RightDiv } from './styled';
 
-const NavBar = ({ onVideosChanged }) => {
-    const [searchValue, setSearchValue] = useState('elon musk on TED Talks');
-    const [darkMode, setDarkMode] = useState(true);
+const NavBar = () => {
+    const { state, dispatch } = useGlobal();
+    const { searchKeyword, isDarkMode } = state;
+    const [darkMode, setDarkMode] = useState(isDarkMode);
+    const [url, setUrl] = useState('');
+    useFetch(url, 'updateVideos');
 
     async function searchVideos(keystroke) {
         if (keystroke === 'Enter') {
-            let response = await fetch(`${API_BASE_URL}?part=snippet&maxResults=25&q=${searchValue}&key=${process.env.REACT_APP_API_KEY}`);
-
-            if (response.ok) { // if HTTP-status is 200-299
-                let json = await response.json();
-                onVideosChanged(json.items);
-            } else {
-                console.log(`HTTP-Error: ${response.status}`);
-            }
+            setUrl(`${API_BASE_URL}?part=snippet&maxResults=25&q=${encodeURI(searchKeyword)}&key=${process.env.REACT_APP_API_KEY}`);
         }
     }
 
+    function updateKeyword(e) {
+        return dispatch({ type: 'search', payload: e.target.value });
+    }
+
+    function toggleTheme() {
+        setDarkMode(!darkMode);
+        dispatch({ type: 'theme', payload: !darkMode });
+    }
```

Comment:
If you decide to use the global state only, then you have to update this one too
```suggestion
    function toggleTheme() {
        dispatch({ type: 'theme', payload: !isDarkMode });
    }
```